lib/Doctrine/DBAL/Connection.php

```diff
@@ -1197,176 +1197,63 @@ public function transactional(Closure $func)
      */
     public function setNestTransactionsWithSavepoints($nestTransactionsWithSavepoints)
     {
-        if ($this->_transactionNestingLevel > 0) {
-            throw ConnectionException::mayNotAlterNestedTransactionWithSavepointsInTransaction();
-        }
-
-        if ( ! $this->getDatabasePlatform()->supportsSavepoints()) {
-            throw ConnectionException::savepointsNotSupported();
-        }
-
-        $this->_nestTransactionsWithSavepoints = (bool) $nestTransactionsWithSavepoints;
+        $this->transactionManager->setNestTransactionsWithSavepoints($nestTransactionsWithSavepoints);
     }
 
     /**
      * Gets if nested transactions should use savepoints.
      *
+     * @deprecated Use getTransactionManager()->getNestTransactionsWithSavepoints()
+     *
      * @return bool
      */
     public function getNestTransactionsWithSavepoints()
     {
-        return $this->_nestTransactionsWithSavepoints;
-    }
-
-    /**
-     * Returns the savepoint name to use for nested transactions are false if they are not supported
-     * "savepointFormat" parameter is not set
-     *
-     * @return mixed A string with the savepoint name or false.
-     */
-    protected function _getNestedTransactionSavePointName()
-    {
-        return 'DOCTRINE2_SAVEPOINT_'.$this->_transactionNestingLevel;
+        return $this->transactionManager->getNestTransactionsWithSavepoints();
     }
 
     /**
      * Starts a transaction by suspending auto-commit mode.
      *
+     * @deprecated Use TransactionManager::beginTransaction()
```

Comment:
It's weird that a function is deprecated in the wrapper connection but not in driver connections which implement the same connection interface. If a caller depends on the interface, how does it know if it calls a deprecated implementation or not?

Reply:
That's why [I questioned whether DBAL\Connection should implement Driver\Connection at all](https://github.com/doctrine/dbal/pull/634#issuecomment-369642037). @stof [seemed to be OK to remove this](https://github.com/doctrine/dbal/pull/634#issuecomment-369648715), and IMO this is the best thing to do.

Note that this is not covered by any tests (removing `implements DriverConnection` passes all the tests), and I can't think of a use case for a common interface for Connection and Driver Connection. I suspect someone just made Connection implement DriverConnection because they could :)

Reply:
Different interfaces for different levels of abstractions make sense. It aligns with the point #_3 in https://github.com/doctrine/dbal/pull/634#issuecomment-370284035: on the higher level, a transaction can be represented with a higher level API object, while on the driver level it's just a driver API call or a SQL statement execution.

Still, as a final solution, deprecation of a method which implements a non-deprecated interface method doesn't make sense.

Reply:
OK, I will definitely remove the `implements DriverConnection` then. `Connection` will use `DriverConnection` under the hood but will not share any interface with it.